acceptance/testdata/restorer/Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM ubuntu:bionic
+
+ARG cnb_uid=1234
+ARG cnb_gid=1000
+
+COPY ./container/ /
+
+# this is required because git on windows does not support our naming of sha256:sha.tar :(
+COPY ./container/cache/committed/29acd01e6fba0963e796f63eb3b517c2b2c6fb2ebcca5180f2bdda82c8b922d0.tar /cache/committed/sha256:29acd01e6fba0963e796f63eb3b517c2b2c6fb2ebcca5180f2bdda82c8b922d0.tar
```

Comment:
I'm totally open to any suggestions on making this less gross. Windows can't check out a file named `sha256:<sha>`, so I am currently just copying it over with the correct name.

Reply:
[Thread](https://buildpacks.slack.com/archives/CSFE49VGR/p1611336072013100)

Reply:
I can't think of how to make this less gross :(

I wonder though if instead of building the cache into the container we could just mount it in - then it would work more like `pack`. Also if we wanted to test cache image or cache directory not available we'd already have a restore image that works for that.

Reply:
I did actually try that. I didn't dig too deep into why it fails though, but it doesn't like the `sha256:blah` naming when using `docker`'s `--volume` support. There is a chance the `--mount` syntax may work better though...

```
» docker run -it --volume "/Users/jesse.brown/dev/buildpacks/lifecycle/acceptance/testdata/restorer/container/cache/committed/29acd01e6fba0963e796f63eb3b517c2b2c6fb2ebcca5180f2bdda82c8b922d0.tar:/cache/committed/sha256:test.tar" heroku/buildpacks:20 bash
docker: Error response from daemon: invalid mode: test.tar.
See 'docker run --help'.
```

Reply:
Could you mount the entire cache directory to avoid having to specify the tar name?

Reply:
The original problem this solves is that we can't have the `sha256:abc.tar` names in our git repo at all, because of windows. So it has to be done in the CLI or the Dockerfile, something that could support that naming scheme.

Reply:
If we generate the tars on the fly though, this problem goes away : )

Reply:
Example of how we mount the cache directory for analyzer acceptance: https://github.com/buildpacks/lifecycle/blob/ee2fd58e6e5c16de1b450f4ddcf819849c6a6d5c/acceptance/analyzer_test.go#L455-L462